src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,854 @@
+import AbstractAlgebra: MatElem, matrix, MatSpace, parent_type, Ring, RingElem
+import Hecke: base_ring, det, fq_nmod, FqNmodFiniteField, nrows, tr, trace
+import GAP: FFE
+
+export
+    general_linear_group,
+    matrix_group,
+    MatrixGroup,
+    MatrixGroupElem,
+    omega_group,
+    orthogonal_group,
+    special_linear_group,
+    special_orthogonal_group,
+    special_unitary_group,
+    symplectic_group,
+    unitary_group,
+    GL, GO, GU, SL, SO, Sp, SU
+
+
+
+abstract type AbstractMatrixGroupElem <: GAPGroupElem{GAPGroup} end
+
+# NOTE: always defined are deg, ring and at least one between { X, gens, descr }
+# NOTE: the fields ring_iso and mat_iso are always defined if the field X is
+"""
+    MatrixGroup{RE<:RingElem, T<:MatElem{RE}} <: GAPGroup
+Type of groups `G` of `n x n` matrices over the ring `R`, where `n = degree(G)` and `R = base_ring(G)`.
+
+At the moment, only rings of type `FqNmodFiniteField` are supported.
+"""
+mutable struct MatrixGroup{RE<:RingElem, T<:MatElem{RE}} <: GAPGroup
+   deg::Int
+   ring::Ring
+   X::GapObj
+   gens::Vector{<:AbstractMatrixGroupElem}
+   descr::Symbol                       # e.g. GL, SL, symbols for isometry groups
+   ring_iso::GenRingIso
+   mat_iso::GenMatIso
+   AbstractAlgebra.@declare_other
+
+   MatrixGroup(m::Int, F::Ring) = new{elem_type(F), dense_matrix_type(elem_type(F))}(m,F)
+
+end
+
+# build a MatrixGroup given the underlying GAP object
+function MatrixGroup(m::Int, F::S, G_gap::GapObj) where S<:Ring
+   r = MatrixGroup(m,F)
+   r.X = G_gap
+   return r
+end
+
+# build a MatrixGroup given a list of generators
+function MatrixGroup(m::Int, F::S, L::Vector) where S<:Ring
```

Comment:
```suggestion
function MatrixGroup(m::Int, F::Ring, L::Vector)
```
In addition, shouldn't the type of `L` be restricted further e.g. to `Vector{<:AbstractMatrixGroupElem}` ?